py4cast/lightning_module/dummy.py

```diff
@@ -0,0 +1,178 @@
+import lightning.pytorch as pl
+import torch
+import torch.nn as nn
+import torchmetrics as tm
+from pathlib import Path
+from typing import Union
+
+# A GARDER
+from py4cast.settings import HIERA_MASKED_DIR
+from py4cast.lightning_module.utils import InitModelLightningModule, PlotLightningModule
+
+# PAS CLAIR
+from copy import deepcopy
+from transformers import get_cosine_schedule_with_warmup
+
+
+# learning rate scheduling period in steps (update every nth step)
+LR_SCHEDULER_PERIOD: int = 10
+
+
+class DummyLightningModule(
+    InitModelLightningModule, PlotLightningModule, pl.LightningModule
+):
+    """A lightning module adapted for test.
+    Computes metrics and manages logging in tensorboard."""
+
+    def __init__(
+        self,
+        model_conf: Union[Path, None] = None,
+        model_name: str = "hiera",
+        lr: float = 0.1,
+        loss_name: str = "mse",
+        len_train_loader: int = 1,
+        save_path: Path = None,
+        use_lr_scheduler: bool = False,
+    ):
+        super().__init__(
+            model_name=model_name, model_conf=model_conf
+        )  # Initialize InitModelLightningModule
+        super(
+            InitModelLightningModule, self
+        ).__init__()  # Initialize PlotLightningModule
+        super(
+            PlotLightningModule, self
+        ).__init__()  # Initialize InitModelLightningModule
+
+        # init model
+        self.model, self.model_settings = self.setup("train")
+
+        self.model_conf = model_conf
+        self.model_name = model_name
+        self.lr = lr
+        self.loss_name = loss_name
+        self.len_train_loader = len_train_loader
+        self.save_path = save_path
+        self.use_lr_scheduler = use_lr_scheduler
+
+        self.training_step_metrics_loss = []
+        self.validation_step_metrics_loss = []
+        self.test_step_metrics_loss = []
+
+        # For making restoring of optimizer state optional (slight hack)
+        self.opt_state = None
+        self.loss = nn.MSELoss(reduction="none")
+        self.metrics = self.get_metrics()
+
+    ###--------------------- MISCELLANEOUS ---------------------###
+
+    def get_metrics(self):
```

Comment:
Better using the `MetricCollection`class provided by torchmetrics : [MetricCollection](https://lightning.ai/docs/torchmetrics/stable/pages/overview.html#metriccollection)